### PR TITLE
SONARJS-901 Disable highlighting, CPD, coverage and metrics in SonarLint context

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptSquidSensor.java [hunk 1]
```diff
@@ -341,7 +335,70 @@ public void execute(SensorContext context) {
...
+    public void executeCoverageSensors() {
+      // unnecessary in SonarLint context
+    }
+  }
+
+  private static boolean isSonarLint(SensorContext context) {
+    return context.getSonarQubeVersion().isGreaterThanOrEqual(V6_0) && context.runtime().getProduct() == SonarProduct.SONARLINT;
   }
 
   private static void executeCoverageSensors(SensorContext context, Map<InputFile, Set<Integer>> linesOfCode, boolean isAtLeastSq62) {
```
Comment (SonarTech) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptSquidSensor.java]: MINOR Move this method into "SonarQubeContextExecutor". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3398)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptSquidSensor.java [hunk 2]
```diff
@@ -341,48 +335,111 @@ public void execute(SensorContext context) {
...
+        isAtLeastSq62);
+      return Arrays.asList(metricsVisitor, new HighlighterVisitor(context), new CpdVisitor(context));
     }
 
-    if (isAtLeastSq62) {
-      logDeprecationForReportProperty(settings, JavaScriptPlugin.LCOV_UT_REPORT_PATH);
-      logDeprecationForReportProperty(settings, JavaScriptPlugin.LCOV_IT_REPORT_PATH);
+    @Override
+    public void executeCoverageSensors() {
+      if (metricsVisitor == null) {
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptSquidSensor.java]: @janos-ss what is the point of this check?

**Comment by vilchik-elena** [comment]
@janos-ss LGTM! Thanks!
I've added some improvements, have a look if you want. 
As it's green, i will merge

**Pull request merged**